metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -1297,3 +1301,64 @@ def visit_semi_additive_join_node(self, node: SemiAdditiveJoinNode) -> SqlDataSe
                 order_bys=(),
             ),
         )
+
+    def visit_join_to_time_spine_node(self, node: JoinToTimeSpineNode[SourceDataSetT]) -> SqlDataSet:  # noqa: D
+        parent_data_set = node.parent_node.accept(self)
+        parent_alias = self._next_unique_table_alias()
+
+        # Build time spine dataset
+        metric_time_dimension_instance: Optional[TimeDimensionInstance] = None
+        for instance in parent_data_set.metric_time_dimension_instances:
+            if len(instance.spec.identifier_links) == 0:
+                metric_time_dimension_instance = instance
+                break
```

Comment:
This returns whatever granularity happens to be first. I think this is usually DAY but we don't have any guarantees around it.

We probably want to do something similar to visit_constrain_time_range_node, because I think we want the time spine comparison to be against the smallest granularity metric time dimension:

https://github.com/transform-data/metricflow/blob/main/metricflow/plan_conversion/dataflow_to_sql.py#L1007-L1030

Reply:
We should update this before merge as well.

Reply:
Oops, missed this before!

Reply:
Just updated to loop through all of them and use the one with the lowest granularity.